### PR TITLE
plot of footprints for cfDNA and scATAC

### PR DESCRIPTION
**Pull request by nrosed**
This PR is 2 main analyses, where I plot the footprints from cfDNA and scATAC-Seq for various transcription factors.
The analyses generally follow a Seurat tutorial but are slightly modified.

The two files to review are: 

1. plot_cfDNA_scATAC/sc_cfDNA_demo.md
2. plot_cfDNA_scATAC/scATAC_demo.md

### CODE REVIEW

File: plot_cfDNA_scATAC/sc_cfDNA_demo.Rmd [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+require(TFBSTools)
+
+
+proj_dir = here()
+
+```
+
+## Overview
+
+This file is an augmented version of the scATAC-seq analysis found here: https://satijalab.org/signac/articles/pbmc_vignette.html
```
Comment (arielah) [inline plot_cfDNA_scATAC/sc_cfDNA_demo.Rmd:42]: Remind me what the point of this repo is? Are you learning the basics of scATAC-seq or is this to help the undergrads?

File: plot_cfDNA_scATAC/scATAC_demo.Rmd [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+pbmc$high.tss <- ifelse(pbmc$TSS.enrichment > 2, 'High', 'Low')
+TSSPlot(pbmc, group.by = 'high.tss') + NoLegend()
+
+
+# plot the fragment periodicity
+# also group the nucleosomal banding pattern
+pbmc$nucleosome_group <- ifelse(pbmc$nucleosome_signal > 4, 'NS > 4', 'NS < 4')
+FragmentHistogram(object = pbmc, group.by = 'nucleosome_group')
+
+VlnPlot(
```
Comment (arielah) [inline plot_cfDNA_scATAC/scATAC_demo.Rmd:164]: The headers on the plots aren't legible, is there something you can add to the {r} code on line 152 to spread the plots out?

File: plot_cfDNA_scATAC/scATAC_demo.Rmd [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+```
+
+now match clusters from RNA-Seq to ATAC-Seq
+
+```{r match_rnaseq, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+
+#pbmc <- subset(pbmc, idents = 14, invert = TRUE)
+pbmc <- RenameIdents(
+  object = pbmc,
+  '0' = 'CD14 Mono',
```
Comment (arielah) [inline plot_cfDNA_scATAC/scATAC_demo.Rmd:307]: Will these clusterings always be the same? Is there a chance the ordering will change?

File: plot_cfDNA_scATAC/scATAC_demo.Rmd [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+  'CD8 Naive' = 'T-cells',
+  'NK CD56Dim' = 'NK',
+  'pre-B' = 'B-cells',
+  'CD16 Mono' = 'Mono',
+  'pro-B' = 'pro-B',
+  'DC' = 'DC',
+  'NK CD56bright' = 'NK',
+  'pDC' = 'DC'
+)
+pbmc <- RenameIdents(
```
Comment (arielah) [inline plot_cfDNA_scATAC/scATAC_demo.Rmd:339]: I'm confused what you're doing here. Are you compressing 'CD4 Naive' to 'T-cells' and then to 'Other'?

File: plot_cfDNA_scATAC/scATAC_demo.Rmd [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+theme_set(theme_bw())
+library(signal)
+library(zoo)
+
+gg_footprint = ggplot(footprint_df, aes(x=position, y=norm.value, color=group)) + 
+                #geom_line() +
+                geom_line(aes(y=rollmean(norm.value, 100, na.pad=TRUE))) +
+                theme_bw() + theme(legend.position = "none", aspect.ratio = 9/16) +
+                labs(x ="Position Along TFBS", y = "Normalized Count")
+gg_footprint
```
Comment (arielah) [inline plot_cfDNA_scATAC/scATAC_demo.Rmd:422]: This could use a legend for what the colors mean.

**Review by arielah — APPROVED** [review]
Looks good! Just a little more info on the purpose of the different analyses would help.

File: plot_cfDNA_scATAC/sc_cfDNA_demo.Rmd (hunk 1 above)
Comment (nrosed) [reply plot_cfDNA_scATAC/sc_cfDNA_demo.Rmd:42]: This is for me to learn and have a reference for how to use Seurat+Signac on cell-free DNA data. So while the vignette focuses on scATAC-Seq data, I have augmented it in small ways to work with cell-free DNA data. I added a blurb of this in https://github.com/greenelab/cfDNA_atac_demo/pull/3/commits/0718bf4ab4e90ec1695652682ec95f955035df9c

File: plot_cfDNA_scATAC/scATAC_demo.Rmd (hunk 2 above)
Comment (nrosed) [reply plot_cfDNA_scATAC/scATAC_demo.Rmd:164]: oo yes, fixed in: https://github.com/greenelab/cfDNA_atac_demo/pull/3/commits/5c7206761b79b44ef1b9318ae4a7d6a087487131

File: plot_cfDNA_scATAC/scATAC_demo.Rmd (hunk 3 above)
Comment (nrosed) [reply plot_cfDNA_scATAC/scATAC_demo.Rmd:307]: no, thankfully they will always be the same as long as nothing in the clustering or the data changes

File: plot_cfDNA_scATAC/scATAC_demo.Rmd (hunk 4 above)
Comment (nrosed) [reply plot_cfDNA_scATAC/scATAC_demo.Rmd]: eek yeah sorry, this is sloppy. I was going back and forth trying to figure out which labeling was better. Clarified here: https://github.com/greenelab/cfDNA_atac_demo/pull/3/commits/e1b7e639f576324543ec6a2d47756582a56f2758

File: plot_cfDNA_scATAC/scATAC_demo.Rmd (hunk 5 above)
Comment (nrosed) [reply plot_cfDNA_scATAC/scATAC_demo.Rmd:387]: ah, yes, added here: https://github.com/greenelab/cfDNA_atac_demo/pull/3/commits/e1b7e639f576324543ec6a2d47756582a56f2758

**Pull request merged**